src/Pawn.cpp

```diff
@@ -0,0 +1,30 @@
+#include "Pawn.h"
+
+Pawn::Pawn(bool isWhite)
+	:Piece(isWhite)
+{
+}
+
+bool Pawn::isCorrectMove(const Location& source, const Location& destination) const
+{
+    if (m_firstMove)
+        if (this->getColor())
+            return (destination.x == source.x + 2 || destination.x == source.x + 1) && destination.y == source.y;
+        else
+            return (destination.x == source.x - 2 || destination.x == source.x - 1) && destination.y == source.y;
+    else
+        if (this->getColor())
+            return destination.x == source.x + 1 && destination.y == source.y;
+        else
+            return destination.x == source.x - 1 && destination.y == source.y;
+}

```

Comment:
function can be simplified by removing the nested if-else statements and combining the conditions. You can also use the ternary operator to make the code more concise. Additionally, consider breaking down the logic into smaller functions to improve readability and maintainability.